skill-tree /src/main/java/com/RDS/skilltree/Skill/SkillDTO.java

```diff
@@ -0,0 +1,28 @@
+package com.RDS.skilltree.Skill;
+
+import com.RDS.skilltree.User.UserModel;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.Builder;
+import lombok.Getter;
+
+import java.util.Set;
+import java.util.UUID;
+
+@Getter
+@Builder
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class SkillDTO {
+    private UUID id;
+    private SkillType type;
+    private String name;
+    private Set<UserModel> users;
+
+    public static SkillDTO toDto(SkillModel skillModel) {
```

Comment:
here users field of the SkillDTO is being set to the users field of the SkillModel. This is a potential problem because the users field of the SkillModel is likely to be a collection of User objects, while the users field of the SkillDTO is likely to be a collection of UserDTO objects. This means that the toDto method is returning a SkillDTO object with a field that contains the wrong type of object.

Reply:
No @iamitprakash in both places we are storing the `UserModel` itself and not the `UserDto`

Reply:
Here if we will be storing the DTO in other entity object then we will have to store the UserDTO in `SkillModel`  and `skillDTO` in the `user Model`

Reply:
In the entities we had a seperate table, but in the entities we have implemented to store the skills in the users and skills storing skills @iamitprakash are you aware of this decision?